redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/DefaultRedisClusterClientFactory.java

```diff
@@ -40,45 +37,30 @@
 @Requires(property = RedisSetting.REDIS_URIS)
 @Singleton
 @Factory
-public class DefaultRedisClusterClientFactory {
+public class DefaultRedisClusterClientFactory extends AbstractRedisClusterClientFactory {
 
-    /**
-     * Create the client based on config URIs.
-     * @param config config
-     * @param defaultClientResources default {@link ClientResources}
-     * @return client
-     */
+    @Override
     @Bean(preDestroy = "shutdown")
     @Singleton
     @Primary
-    public RedisClusterClient redisClient(@Primary AbstractRedisConfiguration config, @Primary @Nullable ClientResources defaultClientResources) {
-        List<RedisURI> uris = config.getUris();
-        if (CollectionUtils.isEmpty(uris)) {
-            throw new ConfigurationException("Redis URIs must be specified");
-        }
-        return defaultClientResources == null ? RedisClusterClient.create(uris) : RedisClusterClient.create(defaultClientResources, uris);
+    public RedisClusterClient redisClient(@Primary AbstractRedisConfiguration config,
+                                          @Primary @Nullable ClientResources defaultClientResources,
+                                          @Nullable List<ClientResourcesMutator> mutators) {
+        return super.redisClient(config, defaultClientResources, mutators);
     }
 
-    /**
-     * Establish redis connection.
-     * @param redisClient client.
-     * @return connection
-     */
+    @Override
     @Bean(preDestroy = "close")
     @Singleton
     @Primary
     public StatefulRedisClusterConnection<String, String> redisConnection(@Primary RedisClusterClient redisClient) {
-        return redisClient.connect();
+        return super.redisConnection(redisClient);
```

Comment:
Why move this method to the base class, when all it does is delegate to the received `redisClient`? I don't really see the point. 🤔